integration_tests/src/main/python/hive_write_test.py

```diff
@@ -115,14 +116,14 @@ def test_optimized_hive_ctas_options_fallback(gens, storage_with_opts, spark_tmp
     with_cpu_session(lambda spark: gen_df(spark, gen_list).createOrReplaceTempView(data_table))
     storage, opts = storage_with_opts
     opts_string = ", ".join(["'{}'='{}'".format(k, v) for k, v in opts.items()])
-    fallback_class = "DataWritingCommandExec" if is_before_spark_340() else "ExecutedCommandExec"
+    fallback_class = "ExecutedCommandExec" if is_spark_340_or_later() or is_databricks122_or_later() else "DataWritingCommandExec"
     assert_gpu_fallback_collect(
         lambda spark: spark.sql("CREATE TABLE {} OPTIONS ({}) STORED AS {} AS SELECT * FROM {}".format(
             spark_tmp_table_factory.get(), opts_string, storage, data_table)),
         fallback_class)
 
 @allow_non_gpu('DataWritingCommandExec,ExecutedCommandExec,WriteFilesExec')
-@pytest.mark.skipif(not (is_hive_available() and is_spark_33X()),
+@pytest.mark.skipif(not (is_hive_available() and is_spark_33X() and not is_databricks122_or_later()),
                     reason="Requires Hive and Spark 3.3.X to write bucketed Hive tables")
```

Comment:
I am revisiting this change to see if we need a follow-on issue

Reply:
With 12.2 we do still fallback, as a result of

```
willNotWorkOnGpu: bucketing is not supported
willNotWorkOnGpu: GPU does not currently support the operator class org.apache.spark.sql.catalyst.expressions.HiveHash
willNotWorkOnGpu: GPU does not currently support the operator class org.apache.spark.sql.execution.RDDScanExec
willNotWorkOnGpu: not all expressions can be replaced
willNotWorkOnGpu: not all data writing commands can be replaced
willNotWorkOnGpu: WriteFilesExec can't run on GPU because parent can't run on GPU
```

but the test fails `Part of the plan is not columnar class org.apache.spark.sql.execution.SortExec` so maybe there is a difference in the plan with 12.2

Reply:
I will modify the test to allow SortExec to fall back. Here is the plan showing the SortExec.

```
!Exec <DataWritingCommandExec> cannot run on GPU because not all data writing commands can be replaced
  !Output <InsertIntoHadoopFsRelationCommand> cannot run on GPU because bucketing is not supported
  !Exec <WriteFilesExec> cannot run on GPU because WriteFilesExec can't run on GPU because parent can't run on GPU
    !Exec <SortExec> cannot run on GPU because not all expressions can be replaced
      @Expression <SortOrder> pmod((hive-hash(b#90) & 2147483647), 3) ASC NULLS FIRST could run on GPU
        @Expression <Pmod> pmod((hive-hash(b#90) & 2147483647), 3) could run on GPU
          @Expression <BitwiseAnd> (hive-hash(b#90) & 2147483647) could run on GPU
            ! <HiveHash> hive-hash(b#90) cannot run on GPU because GPU does not currently support the operator class org.apache.spark.sql.catalyst.expressions.HiveHash
              @Expression <AttributeReference> b#90 could run on GPU
            @Expression <Literal> 2147483647 could run on GPU
          @Expression <Literal> 3 could run on GPU
      ! <RDDScanExec> cannot run on GPU because GPU does not currently support the operator class org.apache.spark.sql.execution.RDDScanExec
        @Expression <AttributeReference> a#89 could run on GPU
        @Expression <AttributeReference> b#90 could run on GPU
        @Expression <AttributeReference> c#91 could run on GPU
```

Reply:
I see now that there are two versions of this test - `test_optimized_hive_bucketed_fallback_33X` and `test_optimized_hive_bucketed_fallback` and the latter seems to match DBR 12.2 behavior, so will enable that test for 12,.2 in addition to skipping `test_optimized_hive_bucketed_fallback_33X`.